In [1]:
# Import libraries:
import pandas as pd
import numpy as np
import os

In [7]:
# Read in data:
df = pd.read_csv('../data/rt_data_processed.csv')

# Filter down to rt thresholds 1-5:
df = df[df['rt'] <= 5]

# Split into cons/libs:
df_cons = df[df['poli_affil'] == 'right']
df_libs = df[df['poli_affil'] == 'left']

In [8]:
# Examine first five rows:
df.head()

userid  rt_userid  rt  orig_rating_ego  orig_rating_peer poli_affil
0     785    2199541   1              0.8          1.142857       left
1     785   15464697   1              0.8          0.200000       left
2     785   15907183   1              0.8          0.428571       left
3     785   16471280   1              0.8          0.428571       left
4     785   19186003   1              0.8          0.904762       left

In [10]:
# Total number of unique users (all/cons/libs):
total_users = len(np.unique(df['userid']))
total_cons = len(np.unique(df_cons['userid']))
total_libs = len(np.unique(df_libs['userid']))
print(f'Total users: {total_users} ({total_cons} conservatives and {total_libs} liberals)')

# Total number of ego-alter relationships:
total_connects = len(df)
con_connects = len(df[df['poli_affil'] == 'right'])
lib_connects = len(df[df['poli_affil'] == 'left'])
print(f'Total ego-alter connections: {total_connects} ({con_connects} conservative and {lib_connects} liberal)')

# Total number of retweets:
total_rts = np.sum(df['rt'])
con_total_rts = np.sum(df_cons['rt'])
lib_total_rts = np.sum(df_libs['rt'])
print(f'Total retweets: {total_rts} ({con_total_rts} conservative and {lib_total_rts} liberal)')

Total users: 66693 (25639 conservatives and 41054 liberals)
Total ego-alter connections: 3898327 (1553814 conservative and 2344513 liberal)
Total retweets: 5722470 (2335428 conservative and 3387042 liberal)


For retweet thresholds 1-5, there are 66,693 unique egos (25,639 conservatives and 41,054 liberals) with 3,898,327 unique ego-alter connections (1,553,814 conservative and 2,344,513 liberal) and 5,722,470 total retweets among all connections (2,335,428 conservative and 3,387,042 liberal).